node-admin/src/test/java/com/yahoo/vespa/hosted/node/admin/integration/ContainerTester.java

```diff
@@ -92,13 +93,16 @@ public class ContainerTester implements AutoCloseable {
                                   storageMaintainer, flagSource,
                                   Collections.emptyList(), Optional.empty(), Optional.empty(), clock, Duration.ofSeconds(-1),
                                   VespaServiceDumper.DUMMY_INSTANCE) {
+                    AtomicBoolean stopped = new AtomicBoolean();
                     @Override public void converge(NodeAgentContext context) {
                         super.converge(context);
                         phaser.arriveAndAwaitAdvance();
                     }
                     @Override public void stopForHostSuspension(NodeAgentContext context) {
-                        super.stopForHostSuspension(context);
-                        phaser.arriveAndDeregister();
+                        if (stopped.compareAndSet(false, true)) {
+                            super.stopForHostSuspension(context);
```

Comment:
So `super.stopForHostSuspension()` should be called for each invocation, but phaser.arriveAndDeregister() only on the first invocation per node?

Reply:
Ideally they would just call `arriveAndAwaitAdvance` on each tick, but I believe these agents may be removed, and no longer tick, is that right? In that case, we need to deregister ...

Reply:
Hmm, perhaps the problem is rather that they `tick` after being removed here ... Urgh.

Reply:
Not sure why we have deregister on `stopForHostSuspension()`. If you want it at as the last call, put it inside `stopForRemoval()`. After that `tick()` will never be called again.

Reply:
Yes, I must have thought suspension was removal when I added this synchronisation some time back. It looks right to just arrive as usual when stopping for suspension. Done now.

Reply:
Node agent instances are normally preserved with stopForHostSuspension() and until jdisc deconstruction.  Only under very particular circumstances would a node agent may be removed after it has invoked stopForHostSuspension(), something like: the reason for the reboot no longer exists, and a has been removed from an application. E.g. RebootTest doesn't test for this.